wallet/views.py

```diff
@@ -0,0 +1,34 @@
+from rest_framework.response import Response
+from rest_framework.views import APIView
+from rest_framework import generics
+from transactions.models import Transaction, TransactionOutput
+from transactions.serializers import TransactionModelSerializer, TransactionOutputModelSerializer
+
+
+class TransactionCreateApiView(generics.CreateAPIView):
+    queryset = Transaction.objects.all()
+    serializer_class = TransactionModelSerializer
+
+
+class WalletInfoView(generics.RetrieveAPIView):
+    def get(self, request, *args, **kwargs):
+        addr = kwargs['own_addr']
+        confirmed_transactions = Transaction.objects.filter(mined=True)
+        received = 0
+        sent = 0
+        for tx in confirmed_transactions:
+            for input in tx.inputs.all():
+                if input.own_addr == addr:
+                    sent += input.value
+            for output in tx.outputs.all():
+                if output.own_addr == addr:
+                    received += output.value
```

Comment:
can we avoid this double for-loop?

Reply:
we have to look at all the inputs and outputs of all confirmed transactions... so we go over all confirmed transactions...and for each such transaction over all its inputs and outputs...hence a double loop

to get rid of it, I don't see any other way than storing state in our inputs and outputs whether they belong to a mined transaction or not...this would require a change in our models ... nor do we think that it would bring much performance benefits(if any)  so we have decided to let it stay